### DISCUSSION TITLE
Welcome to fastapi_amis_admin Discussions!

### DISCUSSION THREAD

**Discussion by amisadmin** [amisadmin/fastapi-amis-admin#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MarshallMoler** [comment]
await site.create_db_and_tables()
AttributeError: 'AdminSite' object has no attribute 'create_db_and_tables'
what happend?
how to solve this problems

**Comment by EchoShoot** [comment]
我使用这个框架一段时间了, 感受如下:
1. 没有一个教程, 当时入门感觉太难了, 然后我是通读了一下 amis 的文档, 看了您怎么实现 amis 的, 再然后又看了你在这里进一步的封装后才会用的.(学习成本有点高)
2. 封装的很好, 看有的朋友特别点赞 class 中放 models, 我第一次用的时候也觉得绝妙, 但随着业务越来越大, 发现了我现在的代码逻辑相当复杂, 全在一个类里面.看明白了代码以后知道先是 page, 然后 page 上的多个参数又交代出去给其他函数来处理, 其他函数中的 amis 节点再把参数继续分散出去. (太过集中)

有没有这样的方式, 咱们前后端分离, 把 amis 当作前端, 把路由注册当作后端, 分别放两个文件, 这样用户可以双开文件,左边写前端界面逻辑, 右边写路由. 可能像这样的结构
```
\- Articles
\- Blog
    |-__init__.py  # 在这里对下面三种组合
    |- models.py  # 用来放模型
    |- views.py   # 用来放路由逻辑
    |- amis.py   # 用来放amis逻辑
```
这样后续推广的时候也比较清晰, 方便大家理解, amis 就是前端界面交互这一块的逻辑、views 就是后端如何配合的逻辑, models 就是前后端操作的模型对象.

**Reply by paullin8676** [reply]
有人知道ColumnOperation中的buttons是怎么用的，传递的是要action列表，在display_list中怎么把admin传给action呢？
